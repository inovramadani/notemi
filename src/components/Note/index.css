.note {
	height: 150px;
	width: 150px;
	margin: 10px 20px 10px 0;
	padding: 10px;
	font-size: 14px;
	display: inline-block;
	position: relative;
}

.note .icon-delete {
	color: white;
	position: absolute;
	right: 2px;
	top: 2px;
	cursor: pointer;
}

.note-text {
	padding-right: 10px;
	width: 140px;
	height: 130px;
	background-color: transparent;
	border: solid 0px transparent;
	resize: none;
	overflow-y: auto;
	outline: none;
}