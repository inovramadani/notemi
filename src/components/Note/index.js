import React from 'react'
import './index.css'

const Note = (props) => (
	<div className="note" style={{backgroundColor: `light${props.color}`}}>
		<i className="fa fa-times-circle icon-delete" onClick={props.onDelete}/>
		<textarea className="note-text" value={props.text} onChange={props.onChange} /> 
	</div>
)

export default Note