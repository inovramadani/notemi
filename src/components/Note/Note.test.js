import React from 'react'
import Note from './index'
import {shallow, mount, render} from 'enzyme'
import renderer from 'react-test-renderer'

describe('<Note />', () => {
	it('renders Note element correctly', () => {
		const note = renderer.create(<Note color="blue" text="hello" style={{backgroundColor: 'lightblue'}}/>).toJSON()
		expect(note).toMatchSnapshot()
	})
})