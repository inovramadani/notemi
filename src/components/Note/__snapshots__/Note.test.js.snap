// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> renders Note element correctly 1`] = `
<div
  className="note"
  style={
    Object {
      "backgroundColor": "lightblue",
    }
  }
>
  <i
    className="fa fa-times-circle icon-delete"
  />
  <textarea
    className="note-text"
    value="hello"
  />
</div>
`;
